util/test/demos/vk/vk_ray_query.cpp

```diff
@@ -0,0 +1,499 @@
+     */
+    VkDescriptorSet descriptorSet = allocateDescriptorSet(descriptorSetLayout);
+
+    VkWriteDescriptorSetAccelerationStructureKHR asWriteDescriptorInfo = {
+        VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET_ACCELERATION_STRUCTURE_KHR,
+        NULL,
+        1,
+        &tlas,
+    };
+
+    VkWriteDescriptorSet asWriteDescriptorSet = {VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET};
+    asWriteDescriptorSet.dstSet = descriptorSet;
+    asWriteDescriptorSet.dstBinding = 0;
+    asWriteDescriptorSet.descriptorCount = 1;
+    asWriteDescriptorSet.descriptorType = VK_DESCRIPTOR_TYPE_ACCELERATION_STRUCTURE_KHR;
+    asWriteDescriptorSet.pNext = &asWriteDescriptorInfo;
+
+    vkh::updateDescriptorSets(device, {asWriteDescriptorSet}, {});
+
+    while(Running())
+    {
+      {
+        VkCommandBuffer cmd = GetCommandBuffer();
+        vkBeginCommandBuffer(cmd, vkh::CommandBufferBeginInfo());
+        pushMarker(cmd, "Copy AS");
+
+        VkCopyAccelerationStructureInfoKHR copyInfo = {
+            VK_STRUCTURE_TYPE_COPY_ACCELERATION_STRUCTURE_INFO_KHR, NULL, blas, newBlas,
+            VK_COPY_ACCELERATION_STRUCTURE_MODE_CLONE_KHR,
+        };
+        vkCmdCopyAccelerationStructureKHR(cmd, &copyInfo);
+
+        popMarker(cmd);
+        CHECK_VKR(vkEndCommandBuffer(cmd));
+
+        Submit(0, 2, {cmd});
+      }
+
+      VkCommandBuffer cmd = GetCommandBuffer();
+
+      vkBeginCommandBuffer(cmd, vkh::CommandBufferBeginInfo());
+
+      VkImage swapimg =
+          StartUsingBackbuffer(cmd, VK_ACCESS_TRANSFER_WRITE_BIT, VK_IMAGE_LAYOUT_GENERAL);
+
+      vkCmdClearColorImage(cmd, swapimg, VK_IMAGE_LAYOUT_GENERAL,
+                           vkh::ClearColorValue(0.1f, 0.1f, 0.1f, 1.0f), 1,
+                           vkh::ImageSubresourceRange());
+
+      vkCmdBeginRenderPass(
+          cmd, vkh::RenderPassBeginInfo(mainWindow->rp, mainWindow->GetFB(), mainWindow->scissor),
+          VK_SUBPASS_CONTENTS_INLINE);
+
+      vkCmdBindPipeline(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, pipe);
+      vkCmdBindDescriptorSets(cmd, VK_PIPELINE_BIND_POINT_GRAPHICS, layout, 0, 1, &descriptorSet, 0,
+                              VK_NULL_HANDLE);
+      vkCmdSetViewport(cmd, 0, 1, &mainWindow->viewport);
+      vkCmdSetScissor(cmd, 0, 1, &mainWindow->scissor);
+      vkh::cmdBindVertexBuffers(cmd, 0, {trisVertexBuffer.buffer}, {0});
+      vkCmdBindIndexBuffer(cmd, trisIndexBuffer.buffer, 0, VK_INDEX_TYPE_UINT32);
+      vkCmdDrawIndexed(cmd, indexCount, 1, 0, 0, 0);
+
+      vkCmdEndRenderPass(cmd);
+
+      FinishUsingBackbuffer(cmd, VK_ACCESS_TRANSFER_WRITE_BIT, VK_IMAGE_LAYOUT_GENERAL);
+
+      CHECK_VKR(vkEndCommandBuffer(cmd));
+
+      Submit(1, 2, {cmd});
+
+      Present();
+    }
+
+    vkDeviceWaitIdle(device);
+
+    vkDestroyAccelerationStructureKHR(device, newBlas, NULL);
+    vkDestroyAccelerationStructureKHR(device, tlas, NULL);
+    vkDestroyAccelerationStructureKHR(device, blas, NULL);
+    trisIndexBuffer.free();
```

Comment:
These frees are unnecessary - AllocatedBuffer's are tracked in a list and destroyed on shutdown as with other common types like pipelines, descriptor sets, etc allocated through/with the VulkanTest rather than directly.